### DISCUSSION TITLE
vtkcamera transformation

### DISCUSSION THREAD

**Discussion by jsaintauret** [pyvista/pyvista#3823] [Q&A]
Hello,

I came across this video below, and I would like to know if any one of you knows how I could do this kind of predefined camera rotation/transformation ?
https://user-images.githubusercontent.com/52704901/212181574-d7facbf0-5437-4870-a994-1046cd50af6d.mp4

The main idea That I'm trying like to implement is a way to constraint the render camera axis of rotation when I move the mouse without having to click on "ctrl", similar to what maltab does when you move a 3D plot with the mouse

![image](https://user-images.githubusercontent.com/52704901/212182538-9d4b0419-cf7b-4899-9cba-9885f6b63693.png)

thank you in advance

**Comment by banesullivan** [comment]
The closest thing I've seen to this is with the `vtkCameraOrientationWidget`, which we have in `add_camera_orientation_widget()`:

https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_camera_orientation_widget.html


https://user-images.githubusercontent.com/22067021/215536475-1c80f074-1f46-4e7f-a89d-eb60201c6f10.mov

**Reply by jsaintauret** [reply]
Hi Brian,
Thanks for your reply, this is kind of what what I was looking for perfect.
To give you more detail I'm using both vedo and pyvista to build a data processing and visualization tool, I would say that I'm mainly using vedo up to 80-90% of vedo function for my tool and the remaining with Pyvista (when some feature are lacking on vedo).

Regarding my main question about the vtkInteractorstyle render rotation (following mouse event) being more similar to a tool like matlab, somebody on vedo share a link to a complete  custom vtkInteractorstyle style. Here the link toward that discussion where you will see the result on my tool (maybe for those who are looking for something similar).

(https://github.com/marcomusy/vedo/discussions/788#discussioncomment-4696207_)

In the meantime I will take a look of Pyvista `add_camera_orientation_widget` and see how I can implement that feature on my tool.

Thanks again

**Reply by banesullivan** [reply]
Thanks for linking that! It could be a fun challenge to implement a custom interaction style for what you're looking for... though I personally don't have the bandwidth to take that on
